kolibri/plugins/learn/assets/src/state/actions/classesActions.js

```diff
@@ -47,6 +47,8 @@ export function showClassAssignmentsPage(store, classId) {
       currentClassroom: {},
     },
   });
+  // Reset examAttemptLogs, so that it will not merge into another exam.
+  store.dispatch('SET_EXAM_ATTEMPT_LOGS', 'RESET_EXAM_LOGS');
```

Comment:
My approach would be to add a `RESET_EXAM_ATTEMPT_LOGS` mutation that clears the logs when the `showExam` action is called. 
```js
  RESET_EXAM_ATTEMPT_LOGS(state) {
    state.examAttemptLogs = {};
  },  
```
```js
export function showExam(store, examId, questionNumber) {
  store.dispatch('RESET_EXAM_ATTEMPT_LOGS');
  ...
}
```